file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,506 @@
+      "! <p class="shorttext">Changeable</p>
+      "! Changeable
+      changeable  TYPE abap_bool,
+      "! <p class="shorttext">Hidden</p>
+      "! Hidden
+      hidden      TYPE abap_bool,
+      "! <p class="shorttext">Use DDIC</p>
+      "! Use ddic description?
+      "! $default 'X'
+      " (type /aif/use_ddic)
+      use_ddic    TYPE abap_bool,
+      "! <p class="shorttext">Description</p>
+      "! Description
+      " (type /aif/err_lb_description)
+      description TYPE c LENGTH 128,
+      "! <p class="shorttext">Short Text</p>
+      "! Short text
+      " (type  /aif/err_lb_short)
+      short_text  TYPE c LENGTH 10,
+      "! <p class="shorttext">Medium Text</p>
+      "! Medium text
+      " (type /aif/err_lb_medium)
+      medium_text TYPE c LENGTH 20,
+      "! <p class="shorttext">Long Text</p>
+      "! Long text
+      " (type /aif/err_lb_long)
+      long_text   TYPE c LENGTH 40,
+    END OF ty_structure_setting.
+
+  "! <p class="shorttext">Structure Settings</p>
+  "! Structure Settings
+  TYPES ty_structure_settings TYPE STANDARD TABLE OF ty_structure_setting WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">Engines</p>
+    "! Engines
+    BEGIN OF ty_engines,
+      "! <p class="shorttext">Application Engine</p>
+      "! Application engine
+      " (type /aif/appl_engine_id)
+      application_engine TYPE n LENGTH 3,
+      "! <p class="shorttext">Persistency Engine</p>
+      "! Persistency engine
+      " (type /aif/persistency_engine_id)
+      persistency_engine TYPE n LENGTH 3,
+      "! <p class="shorttext">Selection Engine</p>
+      "! Selection engine
+      " (type /aif/selection_engine_id)
+      selection_engine   TYPE n LENGTH 3,
+      "! <p class="shorttext">Logging Engine</p>
+      "! Logging engine
+      " (type /aif/logging_engine_id)
+      logging_engine     TYPE n LENGTH 3,
+    END OF ty_engines.
+
+  TYPES:
+    "! <p class="shorttext">Application Interface</p>
+    "! Application interface
+    BEGIN OF ty_main,
+      "! $required
+      format_version       TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header               TYPE zif_aff_types_v1=>ty_header_60_cloud,
+      "! <p class="shorttext">General Information</p>
+      "! General information
+      "! $required
+      aif_interface        TYPE ty_aif_interface,
+      "! <p class="shorttext">Recipient Assignment</p>
+      "! Recipient assignment
+      "! $required
+      recipient_assignment TYPE ty_recipients,
+      "! <p class="shorttext">Structure Settings</p>
+      "! Structure settings
+      "! $required
+      structure_settings   TYPE ty_structure_settings,
+      "! <p class="shorttext">Engines</p>
+      "! Engines
+      "! $required
+      engines              TYPE ty_engines,
+      "! <p class="shorttext">Key Fields</p>
+      "! Key fields
+      "! $required
+      o_key_fields         TYPE ty_o_key_fields,
```

Comment:
Is there a reason for capsuling the table in `ty_o_key_fields`?

Reply:
This is a leftover from trying out different things. I removed it.